api/tests/Behat/bootstrap/v2/Common/ExpectedResultsTrait.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests\Behat\v2\Common;
+
+trait ExpectedResultsTrait
+{
+    private array $comparisonItems = [];
+
+    public function expectedResultsDisplayed($sectionNumber, $expectedResultArrays, $context)
+    {
+        $xpath = '//dl|//tbody';
+        $topLevelTables = $this->getSession()->getPage()->findAll('xpath', $xpath);
+
+        $sections = [];
+        foreach ($topLevelTables as $tableItem) {
+            $this->comparisonItems = [];
+            if ('dl' == $tableItem->getTagName()) {
+                $xpath = "//div[contains(@class, 'govuk-summary-list__row')]";
+                $listSummaryRowItems = $tableItem->findAll('xpath', $xpath);
+
+                if (count($listSummaryRowItems) > 0) {
+                    $xpath = '//li';
+                    $listItems = $tableItem->findAll('xpath', $xpath);
+
+                    if (count($listItems) > 0) {
+                        $this->getTextItems($listItems);
+                    } else {
+                        $xpath = '//dt|//dd';
+                        $descriptionDataItems = $tableItem->findAll('xpath', $xpath);
+                        $this->getTextItems($descriptionDataItems);
+                    }
+                } else {
+                    $xpath = '//li';
+                    $listItems = $tableItem->findAll('xpath', $xpath);
+
+                    if (count($listItems) > 0) {
+                        $this->getTextItems($listItems);
+                    } else {
+                        $xpath = '//dt|//dd';
+                        $descriptionDataItems = $tableItem->findAll('xpath', $xpath);
+                        $this->getTextItems($descriptionDataItems);
+                    }
+                }
```

Comment:
As we're doing what looks like the same actions regardless of if `$listSummaryRowItems` are more than 0 or not, it looks like this can be collapsed down to:

```
            $xpath = '//li';
            $listItems = $tableItem->findAll('xpath', $xpath);
            
            if (count($listItems) > 0) {
                $this->getTextItems($listItems);
            } else {
                $xpath = '//dt|//dd';
                $descriptionDataItems = $tableItem->findAll('xpath', $xpath);
                $this->getTextItems($descriptionDataItems);
            }
```

Unless I'm missing something subtle?